server/src/internalClusterTest/java/org/elasticsearch/health/node/ShardsCapacityHealthIndicatorServiceIT.java

```diff
@@ -88,23 +86,19 @@ private void createIndex(int shards, int replicas) {
     }
 
     private HealthIndicatorResult fetchShardsCapacityIndicatorResult(InternalTestCluster internalCluster) throws Exception {
+        ensureStableCluster(internalCluster.getNodeNames().length);
         var healthNode = ESIntegTestCase.waitAndGetHealthNode(internalCluster);
         assertNotNull(healthNode);
-        var healthNodeName = healthNode.getName();
```

Comment:
@HiDAl I'm not sure why this test was reading the result from the health node. That's not a requirement. The shard capacity indicator does not use any information from the health node (but reads the limits from the cluster state of the node that handles the `_health_report` request)

I've changed the test to reflect this by reading the result from a random node. Happy to make this change in a separate PR if you think it might be problematic or if it doesn't make sense?

Reply:
I think you're right... I copied part of this test from the DiskHealthIndicator, so probably it's a bad copy-paste